packages/protocol/test/ACE/NoteRegistry/epochs/201907/Proxy.js

```diff
@@ -0,0 +1,56 @@
+/* eslint-disable prefer-destructuring */
+/* global artifacts, expect, contract, beforeEach, it, web3:true */
+// ### External Dependencies
+
+// ### Internal Dependencies
+/* eslint-disable-next-line object-curly-newline */
+const { proofs } = require('@aztec/dev-utils');
+const truffleAssert = require('truffle-assertions');
+
+// ### Artifacts
+const AdminUpgradeabilityProxy = artifacts.require('./noteRegistry/proxies/AdminUpgradeabilityProxy.sol');
+const Behaviour = artifacts.require('./noteRegistry/epochs/201907/Behaviour201907.sol');
+
+
+const { BOGUS_PROOF, JOIN_SPLIT_PROOF } = proofs;
+
+contract('Proxy', (accounts) => {
```

Comment:
We should make time to implement more tests. Some initial ideas:

**Success states**
- Can upgrade the behaviour contract and link it to the proxy

**Failure states**
- Should fail to upgrade if not admin
- Should fail if low level `proxy.delegate()` function called on a non-existant contract

Reply:
Ended up adding `should delegate non-admin call even if fn name exists in proxy`, `should fail to upgrade if not admin`, `should fail to set implementation to 0x0`. The first success case you mentioned is tested in Manager.sol.